libraries/chain/block_log.cpp

```diff
@@ -1024,6 +1077,8 @@ namespace eosio { namespace chain {
       return std::clamp(version, min_supported_version, max_supported_version) == version;
    }
 
+   bool block_log::is_new_version(uint32_t version) { return (version >= pruned_transaction_version ? true:false);}
```

Comment:
You can't do it this way. The version might change between this call and the call to get the block.

Reply:
The logic for mismatch has been moved to block_log so this method is unnecessary.